AsyncDisplayKit/Details/ASRangeController.mm

```diff
@@ -145,6 +149,10 @@ - (void)updateCurrentRangeWithMode:(ASLayoutRangeMode)rangeMode
     _didUpdateCurrentRange = YES;
 
     [self setNeedsUpdate];
+  } else if (_needsRangeUpdate) {
+    // If _needsRangeUpdate is YES, but we don't set didUpdateCurrentRange, it
+    // will be overridden.
+    _didUpdateCurrentRange = YES;
```

Comment:
Should we just *always* set this? Otherwise if we don't and someone calls setNeedsUpdate, our mode will be cleared out.